### PR TITLE
Provide a better error message when the user is using jest mocks.

### PR DESCRIPTION
**Pull request by jimfb**
Provide a better error message when the user is using jest mocks.

### CODE REVIEW

**Comment by jimfb** [comment]
cc @spicyj

**Comment by facebook-github-bot-2** [comment]
@jimfb updated the pull request.

File: src/renderers/shared/stack/reconciler/ReactCompositeComponent.js [hunk 1]
```diff
@@ -40,12 +40,24 @@ StatelessComponent.prototype.render = function() {
...
-      'returned undefined, an array or some other invalid object.',
-      Component.displayName || Component.name || 'Component'
-    );
+    var componentName = Component.displayName || Component.name || 'Component';
+    if (element === undefined && Component._isMockFunction) {
+      warning(
+        false,
+        '%s(...): Stateless functional components must return a valid React element, ' +
+        'but %s appears to be a mocked function which returned `undefined`. Consider ' +
+        'disabling mocking (jest.autoMockOff or jest.unmock) for this component or ' +
```
Comment (zpao) [inline src/renderers/shared/stack/reconciler/ReactCompositeComponent.js]: Should we leave "jest" out of this? There are other libraries that people use that do mocking. For example, even you didn't use jest mocking in your test.

**Comment by zpao** [comment]
I just read #6932… not going to do anything further here until a decision is reached there.

**Comment by montogeek** [comment]
Any update about this?

**Comment by gaearon** [comment]
Jest doesn't automock anymore. Is this still useful? I'd imagine that if you mock a React component intentionally, you know what its contract is.

**Comment by sophiebits** [comment]
If you do .mock('Foo') without specifying what to mock it with, I think that mock is still automatically generated. But I agree it's easy to work around.

**Comment by gaearon** [comment]
I'll just close this. If someone feels strongly please send a new PR.

**Pull request closed without merging**